agent/hasalTask.py

```diff
@@ -213,6 +264,10 @@ def run(self):
 
         # touch finish status
         self.touch_status_file(self.DEFAULT_AGENT_JOB_STATUS['FINISH'])
+        if not self.log_fh.closed:
```

Comment:
There are three part have the same code, here, `onstop()`, and `teardown()`.
For reusability, we can extract that to a method.